Python/marshal.c

```diff
@@ -503,9 +503,41 @@ w_complex_object(PyObject *v, char flag, WFILE *p)
             W_TYPE(TYPE_SET, p);
         n = PySet_GET_SIZE(v);
         W_SIZE(n, p);
+        // bpo-37596: We need to write the elements in a deterministic order!
```

Comment:
Try the comment wording suggested above:

The word "deterministic" is correct but doesn't help motivate the algorithm to someone who hasn't read the tracker issue. I recommend:

> To support reproducible builds, sets and frozensets need to be stored with the keys in consistent order even when they have been scrambled by hash randomization. Here we use an order equivalent to: sorted(sorted(s, key=marshal.dumps)). See bpo-37596 for details.